File: typescript/type/cache.ts
/* eslint-disable @typescript-eslint/naming-convention */
import C_Target from "../script/class.js";
import * as _support from "./support.js";
import * as _style from "./style.js";
import * as _config from "./config.js";
import * as _file from "./file.js";

export interface ROOT {
    bin: string,
    name: string,
    version: string,
    vendors: string[],
    extension: string,
    scripts: Record<string, string>,
    commands: Record<string, string>,
    Tweaks: _config.Tweaks,
    customAtrules: Record<string, string>
    customElements: Record<string, number>,
    customOperations: Record<string, string>
    url: {
        Cdn: string,
        Site: string,
        Worker: string,
        Console: string,
        Prefixes: string,
        Artifacts: string,
    }
}

export interface PREFIX {
    atrules: Record<string, Record<string, string>>,
    attributes: Record<string, Record<string, string>>,
    pseudos: Record<string, Record<string, string>>,
    classes: Record<string, Record<string, string>>,
    elements: Record<string, Record<string, string>>,
    values: Record<string, Record<string, Record<string, string>>>,
}

export interface DELTA {
    DeltaPath: string,
    DeltaContent: string,
    PublishError: string,
    FinalMessage: string,
    ErrorCount: number,
    Report: {
        artifacts: string,
        libraries: string,
        archives: string,
        constants: string,
        hashrule: string,
        errors: string,
        memChart: string,
        footer: string,
    },
    Lookup: {
        artifacts: Record<string, _file.Lookup>,
        libraries: Record<string, _file.Lookup>,
        archives: Record<string, _file.Lookup>,
    },
    Errors: {
        artifacts: string[],
        libraries: string[],
        archives: string[],
        multiples: string[],
    },
    Diagnostics: {
        multiples: _support.Diagnostic[],
        artifacts: _support.Diagnostic[],
        libraries: _support.Diagnostic[],
        archives: _support.Diagnostic[],
    },
    Manifest: {
        constants: string[],
        hashrules: Record<string, string>,
        filelookup: Record<string, _file.Lookup>,
        AXIOM: Record<string, _file.ClassMetaMap>,
        CLUSTER: Record<string, _file.ClassMetaMap>,
        LOCAL: Record<string, _file.ClassMetaMap>,
        GLOBAL: Record<string, _file.ClassMetaMap>,
        ARTIFACT: Record<string, _file.ClassMetaMap>,
        errors: _support.Diagnostic[]
    },
}

export interface STATIC {
    WATCH: boolean,
    DEBUG: boolean,
    Command: string,
    Argument: string,
    RootCSS: string,
    RootPath: string,
    WorkPath: string,
    ProjectName: string,
    ProjectVersion: string,
    Archive: _config.Archive,
    ProxyMap: _config.ProxyMap[],
    Tweaks: _config.Tweaks,
    Prefix: PREFIX,
    Hashrule: Record<string, string>,
    Artifacts_Saved: Record<string, string>,
    Libraries_Saved: Record<string, string>,
    Targetdir_Saved: Record<string, _config.ProxyStorage>,
}

export interface CLASS {
    Hashrule: Record<string, string>,
    Index_to_Data: Record<string, _style.Classdata>,
    Global___Index: _style.ClassIndexMap,
    Public___Index: _style.ClassIndexMap,
    Library__Index: _style.ClassIndexMap,
    Artifact_Index: _style.ClassIndexMap,
    Sync_ClassDictionary: _style.Dictionary,
    Sync_PublishIndexMap: _style.ClassIndexTrace,
};

export interface FILES {
    LIBRARIES: Record<string, _file.Storage>,
    ARTIFACTS: Record<string, _file.Storage>,
    TARGETDIR: Record<string, C_Target>,
}

File: gosource/types/cache.go
package types

type Cache_Url struct {
	Cdn       string
	Site      string
	Worker    string
	Console   string
	Prefixes  string
	Artifacts string
}

type Cache_Root struct {
	Bin              string
	Name             string
	Version          string
	Extension        string
	Scripts          map[string]string
	Commands         map[string]string
	Tweaks           Config_Tweaks
	CustomAtrules    map[string]string
	CustomElements   map[string]int
	CustomOperations map[string]rune
	Url              Cache_Url
}

type Cache_Prefix struct {
	Atrules    map[string]map[string]string            `json:"atrules"`
	Attributes map[string]map[string]string            `json:"attributes"`
	Pseudos    map[string]map[string]string            `json:"pseudos"`
	Classes    map[string]map[string]string            `json:"classes"`
	Elements   map[string]map[string]string            `json:"elements"`
	Values     map[string]map[string]map[string]string `json:"values"`
}

type Cache_Static struct {
	WATCH           bool
	DEBUG           bool
	Command         string
	Argument        string
	RootCSS         string
	RootPath        string
	WorkPath        string
	ProjectName     string
	ProjectVersion  string
	Vendors         []string
	ProxyMap        []Config_ProxyMap
	Tweaks          Config_Tweaks
	Hashrule        map[string]string
	Artifacts_Saved map[string]string
	Libraries_Saved map[string]string
	Targetdir_Saved map[string]Config_ProxyStorage
	Archive         Config_Archive
}

type Cache_Delta_Report struct {
	Artifacts string
	Libraries string
	Archives  string
	Constants string
	Hashrule  string
	Errors    string
	MemChart  string
	Footer    string
}

type Cache_Delta_Lookup struct {
	Libraries map[string]File_Lookup
	Artifacts map[string]File_Lookup
	Archives  map[string]File_Lookup
}

type Cache_Delta_Errors struct {
	Artifacts []string
	Libraries []string
	Archives  []string
	Multiples []string
}

type Cache_Delta_Diagnostics struct {
	Multiples []Support_Diagnostic
	Artifacts []Support_Diagnostic
	Libraries []Support_Diagnostic
	Archives  []Support_Diagnostic
}

type Cache_Delta struct {
	DeltaPath    string
	DeltaContent string
	PublishError string
	FinalMessage string
	ErrorCount   int
	Report       Cache_Delta_Report
	Lookup       Cache_Delta_Lookup
	Errors       Cache_Delta_Errors
	Diagnostics  Cache_Delta_Diagnostics
}

type Cache_Manifest struct {
	Constants  []string
	Hashrule   map[string]string
	Filelookup map[string]File_Lookup
	AXIOM      map[string]File_ClassMetaMap
	CLUSTER    map[string]File_ClassMetaMap
	LOCAL      map[string]File_ClassMetaMap
	GLOBAL     map[string]File_ClassMetaMap
	ARTIFACT   map[string]File_ClassMetaMap
	Errors     []Support_Diagnostic
}

type Cache_Class struct {
	Hashrule        map[string]string
	Index_to_Data   map[int]Style_ClassData
	Global___Index  Style_ClassIndexMap
	Public___Index  Style_ClassIndexMap
	Library__Index  Style_ClassIndexMap
	Artifact_Index  Style_ClassIndexMap
	ClassDictionary Style_Dictionary
	PublishIndexMap []Style_ClassIndexTrace
}

type Cache_Files struct {
	LIBRARIES map[string]File_Storage
	ARTIFACTS map[string]File_Storage
	TARGETDIR map[string]Script_Target
}
